actor/builtin/miner/miner_test.go

```diff
@@ -490,6 +493,120 @@ func TestMinerSubmitPoStSectorSetUpdates(t *testing.T) {
 	require.NoError(t, res.ExecutionError)
 	require.Equal(t, uint8(0), res.Receipt.ExitCode)
 
+	t.Run("power is 0 until first submitPoSt", func(t *testing.T) {
```

Comment:
It's a little weird to include a bunch of subtests in a test that depend on each other. I usually expect subtests to be independent. I think writing tests that assert a sequence of things is fine (though some disagree). I would:

1. Remove the subtests and replace the descriptions with comments.
2. Create a test struct with st, vms, and minerAddress.
3. Give it a methods that can call submitPoSt or commitSector so that it accept the message parameters, runs the message and then return the miner's power and the total storage power.
4. Replace the 7 calls in this class with that method.

That should make this test pretty short and easy to understand.

Reply:
Thanks, this is all great advice.  Here are some meta-notes

I copied the "sequential dependent subtest" pattern from TestMinerSubmitPoSt.  I assumed this was the way things were done in miner actor testing land and leaned on this structure to get the new tests written.  The tests I wrote are a good example of how bad habits multiply when new people jump into half-baked testing packages without realizing they are incomplete.

Your PR comment is a good example of what we need to do to prevent these bad habits from ballooning.

In general this project suffers from a lack of reusable quality test frameworks.  1) because the emergent test frameworks are low quality (like the one I just copied) they are burdensome to use to write new tests if you are new to the code. 2) the burden to improve these emergent frameworks falls on people implementing features which slows us all down.